cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -540,6 +553,13 @@ public void createApplicationVersion(ApplicationId id,
     if (latest != null) {
       List<Field<?>> fields = getApplicationPrimaryKeys(id.getNamespace(), id.getApplication(),
                                                         latest.getSpec().getAppVersion());
+      // Assign a creation time if it's null
+      // It is for the case that user has a pre-6.8 version, we mark it as past version (like created 1s ago)
+      // So it's sortable on creation time, especially when UI displays the version history for a pipeline
+      if (latest.getChange() == null || latest.getChange().getCreationTimeMillis() == 0) {
+        fields.add(Fields.longField(StoreDefinition.AppMetadataStore.CREATION_TIME_FIELD,
+                                    System.currentTimeMillis() - 1000));
```

Comment:
should use the creation time from appMeta instead of the current time. Otherwise it's actually possible this time is later than the new version. It also makes it deterministic and easy to unit test, as you know exactly what the creation time should be based on the input.

Reply:
Nice catch, updating this